Assets/Scripts/Data Model/Layer.cs

```diff
@@ -1,1450 +1,1481 @@
-	{
-		T entity = (T)subEntity.Entity;
-		//entity.RestoreSubEntity(subEntity);
-		subEntity.RestoreDependencies();
-		int persistentID = subEntity.GetPersistentID();
+			if (entityType == null) { Debug.LogError("Entity type error in " + FileName); }
+			Debug.LogError("Failed to find key from entity type " + entityType.Name + " in " + FileName);
+
+			return 0;
+		}
 
-		if (entity.PlanLayer == currentPlanLayer)
+		public override void RemoveSubEntity(SubEntity subEntity, bool uncreate = false)
 		{
-			//This entity used to be inactive, and is now made active
-			if (entity.GetSubEntityCount() == 1)
+			T entity = (T)subEntity.Entity;
+			subEntity.RemoveDependencies();
+			int persisID = subEntity.GetPersistentID();
+
+			//If it wasnt in the curren't plan, simply add to removed geom
+			if (entity.PlanLayer == currentPlanLayer)
 			{
-				currentPlanLayer.AddNewGeometry(entity);
+				if (entity.GetSubEntityCount() != 1)
+				{
+					//If there are no other subentities left in the entity
+					throw new Exception("Entity has an invalid subentity count. Count: " + entity.GetSubEntityCount() + " on entity with database ID " + entity.DatabaseID);
+				}
+				currentPlanLayer.RemoveNewGeometry(entity);
+				activeEntities.Remove(entity);
 
-				// If another object was being displayed because of the deletion, deactivate it and show this instead
-				if (!subEntity.Entity.Layer.IsPersisIDCurrentlyNew(persistentID))
+				//If the entity wasn't created in this plan, active the last entity with that persisID and add it to removedgeom
+				if (!subEntity.Entity.Layer.IsPersisIDCurrentlyNew(persisID))
 				{
-					if (!recreate) //We are only recreating a newer version, the ID isn't in removedgeom
+					if (!uncreate) //We are uncreating, only reactivate previous version, don't add to removed geom
 					{
-						RemovePreModifiedEntity(subEntity.Entity);
-						entity.PlanLayer.RemovedGeometry.Remove(persistentID);
+						AddPreModifiedEntity(subEntity.Entity);
+						entity.PlanLayer.RemovedGeometry.Add(persisID);
 					}
-					DeactivateCurrentEntityWith(persistentID);
+					ActivateLastEntityWith(persisID);
 				}
-
-				//Only add to activeentities after calling DeactivateCurrentEntityWith
-				activeEntities.Add(entity);
 			}
+			else //Was on another layer and removed by this plan. It's removed and displayed with (-) icons.
+			{
+				currentPlanLayer.RemovedGeometry.Add(persisID);
+				AddPreModifiedEntity(subEntity.Entity);
+			}
+			entity.RedrawGameObjects(CameraManager.Instance.gameCamera);
 		}
-		else //Was on another layer and removed by this plan
+
+		public override void RestoreSubEntity(SubEntity subEntity, bool recreate = false)
 		{
-			currentPlanLayer.RemovedGeometry.Remove(persistentID);
-			RemovePreModifiedEntity(subEntity.Entity);
-		}
-		entity.RedrawGameObjects(CameraManager.Instance.gameCamera);
-	}
+			T entity = (T)subEntity.Entity;
+			//entity.RestoreSubEntity(subEntity);
```

Comment:
Commented code, as always see if it makes sense to leave it there for the future, otherwise remove it! :)